com.unity.cinemachine/Editor/Samples/SampleDependencyImporter.cs

```diff
@@ -76,53 +80,66 @@ void LoadAssetDependencies(string assetPath)
                         if (sampleEntry != null)
                         {
                             // Import common asset dependencies
-                            assetsImported =
-                                ImportAssetDependencies(m_PackageInfo, m_SampleConfiguration.SharedAssetDependencies);
+                            sharedAssetsImported = ImportAssetDependencies(
+                                m_PackageInfo, m_SampleConfiguration.SharedAssetDependencies, 
+                                out bool sharedAssetsReimported);
                             
                             // Import sample-specific asset dependencies
-                            assetsImported |=
-                                ImportAssetDependencies(m_PackageInfo, sampleEntry.AssetDependencies);
+                            localAssetsImported = ImportAssetDependencies(
+                                m_PackageInfo, sampleEntry.AssetDependencies, 
+                                out bool localAssetsReimported);
                             
                             // Import common amd sample specific package dependencies
                             m_PackageDependencyIndex = 0;
                             m_PackageDependencies = new List<string>(m_SampleConfiguration.SharedPackageDependencies);
                             m_PackageDependencies.AddRange(sampleEntry.PackageDependencies);
-                            if (m_PackageDependencies.Count != 0 && PromptUserConfirmation(m_PackageDependencies))
+                            
+                            if (m_PackageDependencies.Count != 0 && 
+                                DoDependenciesNeedToBeImported(out var dependenciesToImport))
                             {
-                                // Import package dependencies using the editor update loop, because
-                                // adding packages need to be done in sequence one after the other
-                                EditorApplication.update += ImportPackageDependencies;
+                                switch (PromptUserConfirmation(dependenciesToImport))
```

Comment:
This code should be before the ImportAssetDependencies part, because if they cancel we don't want to import anything.

Reply:
I think my hack works reliably so we can leave it in.

I modified the code so that X is abort.